x-pack/metricbeat/module/aws/utils.go

```diff
@@ -45,6 +45,10 @@ func GetListMetricsOutput(namespace string, regionName string, svcCloudwatch clo
 
 	listMetricsInput := &cloudwatch.ListMetricsInput{
 		NextToken: nextToken,
+		// To filter the results to show only metrics that have had data points published
+		// in the past three hours, specify this parameter with a value of PT3H.
+		// Please see https://docs.aws.amazon.com/AmazonCloudWatch/latest/APIReference/API_ListMetrics.html for more details.
+		RecentlyActive: types.RecentlyActivePt3h,
```

Comment:
I'm still not convinced we can apply this param with `perdiod > 3h`
https://github.com/elastic/beats/blob/main/x-pack/metricbeat/module/aws/utils.go#L83-L85

`MetricDataQueries: metricDataQueriesPartial` comes from the list of metrics returned from the call above
if we have a period of 24h, and first listing is at 1am 30 Sept, the next listing will be 1am 1 Oct.
Assuming a resource lived between 3pm 30 Sept and 7pm 30 Sept, the list of metrics for the related dimension won't be returned

Reply:
> But with the existing code `FindTimestamp` function in https://github.com/elastic/beats/blob/main/x-pack/metricbeat/module/aws/utils.go#L128 , we are only collecting data points with the last/latest timestamp from the returned `MetricData` list

this `FindTimestamp` and the later `CheckTimestampInArray` is a little obscure to me.
in `FindTimestamp` we don't do any ordering: just return the first between either a single or a couple of timestamps in the `MetricData` list. do we know if they are ordered? and why do we filter by this timestamp later on in `CheckTimestampInArray` (with `time.Time.Equal`!)?

what does exactly the timestamp returned by `FindTimestamp` represent? :)

Reply:
Yeah tbh this is pretty messed up for me too... and I'm finding more and more bugs associated with these functions, such as I suspect https://github.com/elastic/beats/issues/30915 is related... I think we will have a separate refactoring PR coming up soon... 

So timestamps in the `MetricData` list are in order and the timestamp returned by `FindTimestamp` is the common timestamp among all returned metrics. `MetricData` list can return multiple metrics and each metric can include multiple data points. For example `CPUUtilization` with two data points from [5:05, 5:00] and `DiskReadBytes` with one data point from [5:05]. `FindTimestamp` function returns the common timestamp 5:05 between these two and `CheckTimestampInArray` returns which data point is it for this timestamp.

Reply:
> So timestamps in the `MetricData` list are in order and the timestamp returned by `FindTimestamp` is the common timestamp among all returned metrics.

and I see a problem here: all returned metrics for every dimension (ec2 instances, rds instances, etc) not for each specific dimension. but what is the first dimension to be listed?

going back to the `RecentlyActive`:
if `period > 3h` and the first listed dimension is one recently active, yes, the non-recently active will be discarded. but if the first listed dimension is a non-recently active we'll get "stale" data for all the recently active because of what `FindTimestamp` returns

Reply:
> I'd like to leave `FindTimestamp` for a different PR.

oki for me